server/user/mw.js

```diff
@@ -0,0 +1,18 @@
+'use strict';
+
+const crypto = require('crypto');
+
+function getKeyFromRequest(req, res, next) {
+  const key = [req.ip, req.body.email].join(':');
+  req.key = crypto.createHash('sha256').update(key).digest('base64');
```

Comment:
Should we prefix `req.key` with something to reduce the chance of collision and make it easier to read (i.e. `req.userKey`)?